pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1131,11 +1131,11 @@ def _looks_like_infinite_iterator(param: nodes.NodeNG) -> bool:
             return False
 
         if isinstance(node.func, nodes.Attribute):
-            # A next() method, which is now what we want.
+            # Only want the builtin next method.
             return
 
         inferred = utils.safe_infer(node.func)
-        if getattr(inferred, "name", "") == "next":
+        if inferred.qname() == "builtins.next":
```

Comment:
I think we should use an `isinstance` as well, to make sure the `qname` method actually exists.